internal/infrastructure/gateways/datgroup_api_service.go

```diff
@@ -140,21 +140,37 @@ func parseXML(logger *zerolog.Logger, datgroupRespXML, vin string) (*DATGroupInf
 	if err != nil {
 		logger.Err(err).Str("vin", vin).Msgf("failed to extract year low/year for datgroup vin decode")
 	}
+
 	if yearLow > 2000 {
 		response.YearLow = yearLow
 	}
+
 	if yearHigh > 2000 {
 		response.YearHigh = yearHigh
 	}
+
 	yr := shared.VIN(response.VIN).Year()
 	if yr >= response.YearLow && yr <= response.YearHigh {
 		response.Year = yr
 	} else {
 		response.Year = response.YearHigh
 	}
+
 	// series equipment
 	seriesEquipment := xmlquery.FindOne(vehicle, "//ns1:SeriesEquipment")
+
+	if seriesEquipment == nil {
+		logger.Err(err).Str("vin", vin).Msgf("failed to find series equipment node")
```

Comment:
the err here would be from a different error, lets instantiate a new fmt.errorf(..) and then return that as well as log - although check the the caller already logs the error (may not be necessary to log just return nil, the error)